staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1035,13 +1035,15 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 			for index := range toProcess {
 				accessor, err := meta.Accessor(items[index])
 				if err != nil {
-					errs <- err
-					return
+					select {
+					case errs <- err:
+					default:
+					}
+					continue
 				}
 				if _, _, err := e.Delete(ctx, accessor.GetName(), deleteValidation, options); err != nil && !apierrors.IsNotFound(err) {
 					klog.V(4).Infof("Delete %s in DeleteCollection failed: %v", accessor.GetName(), err)
 					errs <- err
-					return
```

Comment:
I haven't read down all the way yet, but the only reason I can think that you didn't build a similar select structure here is that your unit test never checks this branch.

Reply:
I run the new test which hits this line:
```
k8s.io/apiserver/pkg/registry/generic/registry.(*Store).DeleteCollection.func2 at store.go:1061
runtime.goexit at asm_amd64.s:1357
 - Async stack trace
k8s.io/apiserver/pkg/registry/generic/registry.(*Store).DeleteCollection at store.go:1028
```